python/alibiexplainer/alibiexplainer/explainer.py

```diff
@@ -10,38 +10,37 @@
 from alibiexplainer.anchor_tabular import AnchorTabular
 from kfserving.server import Protocol
 from kfserving.protocols.util import NumpyEncoder
-from alibiexplainer.explainer_method import ExplainerMethodImpl
+
 
 logging.basicConfig(level=kfserving.server.KFSERVER_LOGLEVEL)
 
 
 class ExplainerMethod(Enum):
     anchor_tabular = "anchor_tabular"
 
+    def __str__(self):
+        return self.value
 
 class AlibiExplainer(kfserving.KFModel):
     def __init__(self,
                  name: str,
                  predict_url: str,
                  protocol: Protocol,
                  method: ExplainerMethod,
-                 training_data_url: str = None):
+                 config: Mapping,
+                 explainer: object = None):
         super().__init__(name)
         self.predict_url = predict_url
         self.protocol = protocol
         self.method = method
-        self.training_data_url = training_data_url
+
         if self.method is ExplainerMethod.anchor_tabular:
-            self.explainer: ExplainerMethodImpl = AnchorTabular(self._predict_fn)
-            self.explainer.validate(self.training_data_url)
+            self.wrapper = AnchorTabular(self._predict_fn,explainer,**config)
         else:
             raise NotImplementedError
 
     def load(self):
```

Comment:
Does the explainer instantiate immediately?

Reply:
If the saved pickled explainer is provided it will be fast otherwise it will need to do the processing of training data (in this case). However, we should encourage the use of picked pre-trained explainers. Or do you mean something else?

Reply:
SGTM